components/src/utils/Bloomreach.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright © 2019 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+export default function bloomreachFetch(input, requestInit) {
+  return fetch(input, requestInit)
+    .then((res) => {
+      if (res.status >= 500) {
+        if (window.location.href.indexOf('/maintenance') === -1) {
```

Comment:
If this is a general fetch, can you add it to the Cortex.ts and call from there? It would also allow future calls to re-use the same function.

Reply:
hm yeah it is general fetch call and works exactly the same as cortex fetch.  Just didn't seem appropriate using `Cortex.ts` with bloomreach related calls.  Perhaps we could rename Cortex.ts to something general like `FetchInfo.ts` which would still hold cortex 
specific fetch functions but also bloomreach ones.

Reply:
@aChanEP Considering the current work with the cortex client lib, maybe we can keep this for now and revisit if it's needed later. 😄